server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/CreateSnapshotRequest.java

```diff
@@ -94,31 +92,56 @@ public CreateSnapshotRequest(StreamInput in) throws IOException {
         super(in);
         snapshot = in.readString();
         repository = in.readString();
-        indices = in.readStringArray();
-        indicesOptions = IndicesOptions.readIndicesOptions(in);
-        if (in.getVersion().after(Version.V_4_5_1)) {
-            templates = in.readStringArray();
+        String[] oldIndices = Strings.EMPTY_ARRAY;
+        String[] oldTemplates = Strings.EMPTY_ARRAY;
+        if (in.getVersion().before(Version.V_6_0_0)) {
+            oldIndices = in.readStringArray();
+            IndicesOptions.readIndicesOptions(in);
+            oldTemplates = in.readStringArray();
         }
         settings = readSettingsFromStream(in);
         includeGlobalState = in.readBoolean();
         waitForCompletion = in.readBoolean();
         partial = in.readBoolean();
+        HashSet<RelationName> relationNamesSet = new HashSet<>();
```

Comment:
I don't get why we need this logic, shouldn't we stream in an `if/else` manner, so if it's before 6.0.0
use the old stuff or else the new relationNames/partitionNames?

Reply:
We must construct RelationNames from nodes sending 'old' requests containing only indices and templates. The 'new' request only work with RelationNames

Reply:
apologies if I'm confused, I mean: if `in` is before 6.0.0 we only read the old arrays, we don't need to also read the new ones and merge using the set? or I'm missing something?

Reply:
Right, I've simplified this part and I also spotted an issue, partitionNames weren't constructed. Pushed a fixup.